src/class/video/video_device.c

```diff
@@ -960,54 +1023,48 @@ static int handle_video_stm_cs_req(uint8_t rhport, uint8_t stage,
     case VIDEO_VS_CTL_PROBE:
       if (self->state != VS_STATE_PROBING) {
         self->state = VS_STATE_PROBING;
```

Comment:
Also I feel that in case of dynamic setting, we shoudln't change the state to PROBING when it is currently at STREAMING. Though I am not too familliar with UVC, and got a bit tired after tracing down the issue, we can revise this state change later on.

Reply:
Thank you for investigating this issue.

When implementing UVC, I thought that once state transition from PROBING to STREAMING, the buffer used for PROBING becomes unnecessary and can be reused. So, I implemented using the same ep_buf for both PROBING and STREAMING.

Anyway, this issue actually existed and was resolved by your patch. I think that approach is correct.

Reply:
thank you for the review, I thought that as well :)